docs/source/_templates/sdk_TP_tutorial.rst

```diff
@@ -0,0 +1,378 @@
+state of the game (e.g. the board layout and whose turn it is).
+
+{% else %}
+``apply`` gets called with two arguments, ``transaction`` and ``state_store``.
+``transaction`` holds the command that is to be executed (e.g. taking a space or
+creating a game), while ``state_store`` stores information about the current
+state of the game (e.g. the board layout and whose turn it is).
+{% endif %}
+
+Without yet getting into the details of how this information is encoded, we can
+start to think about what ``apply`` needs to do. ``apply`` needs to
+
+1) unpack the command data from the transaction,
+2) retrieve the game data from the state store,
+3) play the game, and
+4) save the updated game data.
+
+Accordingly, a top-down approach to ``apply`` might look like this:
+
+{% if language == 'JavaScript' %}
+
+.. TODO::
+
+    This example code should be rewritten by a JavaScript expert
+    to parallel Python example.
+
+.. code-block:: javascript
+
+    apply (transactionProcessRequest, state) {
+    return _decodeRequest(transactionProcessRequest.payload)
+      .catch(_toInternalError)
+      .then((update) => {
+        let header = TransactionHeader.decode(transactionProcessRequest.header)
+        let player = header.signerPubkey
+        if (!update.name) {
+          throw new InvalidTransaction('Name is required')
+        }
+
+        if (!update.action) {
+          throw new InvalidTransaction('Action is required')
+        }
+
+        // Perform the action
+        let handlerFn
+        if (update.action === 'create') {
+          handlerFn = _handleCreate
+        } else if (update.action === 'take') {
+          handlerFn = _handleTake
+        } else {
+          throw new InvalidTransaction(`Action must be create or take not ${verb}`)
+        }
+
+        let address = XO_NAMESPACE + _hash(update.name)
+
+        return state.get([address]).then(handlerFn(state, address, update, player))
+          .then((addresses) => {
+            if (addresses.length === 0) {
+              throw new InternalError('State Error!')
+            }
+          })
+      })
+    }
+
+{% else %}
+{# Python code is the default #}
+
+.. code-block:: python
+
+    def apply(self, transaction, state_store):
+        signer, game_name, action, space = \
+            self._unpack_transaction(transaction)
+
+        board, state, player1, player2 = \
+            self._get_state_data(game_name, state_store)
+
+        updated_game_data = self._play_xo(
+            board, state,
+            player1, player2,
+            signer, action, space
+        )
+
+        self._store_game_data(game_name, updated_game_data, state_store)
+
+{% endif %}
+
+Note that the third step is the only one that actually concerns tic-tac-toe;
+the other three steps all concern the coordination of data.
+
+Data
+====
+
+So how do we get data out of the transaction? The transaction consists of a
+header and a payload. The header contains the "signer", which is used to
```

Comment:
Since we have the Transaction And Batches! section of the architecture docs, I would provide a link to that here.